### PR TITLE
[Bugfix] Enable pipeline profiling for LingBot Video

### LINKED ISSUE #7741: [Bug][Diffusion][LingBot] LingBotVideoPipeline ignores --enable-diffusion-pipeline-profiler (no Diffuse / stage_durations fine-grained timings)
## Summary

Dense `LingBotVideoPipeline` (`vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py`) does **not** implement `DiffusionPipelineProfilerMixin`. Passing `--enable-diffusion-pipeline-profiler` (e.g. from DFX perf JSON) has no effect: worker logs never emit `[DiffusionPipelineProfiler] ... took ...`, and `/v1/videos` responses only expose orchestrator-level timings (`queue_wait_ms`, `stage_0_gen_ms`) in `stage_durations`.

By contrast, `LingBotWorldCausalDMDPipeline` (`lingbot_world/pipeline.py`) already calls `setup_diffusion_pipeline_profiler(...)` with custom targets (`_generate_block`, `vae.decode`, etc.).

This breaks the expectation of DFX / `vllm bench serve --omni` Stage Durations printing (Diffuse / VAE / TextEncoder-style keys) for the dense LingBot video nightly perf case.

Related: PR https://github.com/vllm-project/vllm-omni/pull/7737 (perf migration + `stage_durations` collection in omni bench).

## Environment

- Model: `robbyant/lingbot-video-dense-1.3b`
- Serve: `--model-class-name LingBotVideoPipeline --enable-diffusion-pipeline-profiler`
- Client: `tests/dfx/perf/tests/test_lingbot_video_vllm_omni.json` via `run_benchmark.py` / `vllm bench serve --omni` (`POST /v1/videos`)

## Repro

```bash
export BENCHMARK_DIR=tests/dfx/perf/results
pytest -s -v tests/dfx/perf/scripts/run_benchmark.py \
  --test-config-file tests/dfx/perf/tests/test_lingbot_video_vllm_omni.json \
  -m "H100 and B200 and cards_1"
```

(Or serve the same model/class with `--enable-diffusion-pipeline-profiler` and call `/v1/videos`.)

## Observed

1. Worker logs: **no** `[DiffusionPipelineProfiler]` lines; only OmniTiming / StageRequestStats (`stage_gen_time_ms` ≈ 160 ms for the tiny 320x192 / 9-frame / 2-step workload).
2. Bench Stage Durations only prints:
   - `Mean Queue Wait (ms)`
   - `Mean Stage 0 Gen (ms)` (matches server `stage_gen_time_ms`)
3. No pipeline fine-grained keys such as `Diffuse`, `Generate Block`, `Vae Decode`, `Text Encoder Forward`.

## Expected

With `--enable-diffusion-pipeline-profiler`:

- Worker logs `[DiffusionPipelineProfiler] <stage> took ...`
- Response `stage_durations` includes pipeline profiler keys (for LingBot dense, at least the hot paths used by offline T2V: e.g. denoise / transformer step, text encode, VAE decode)
- Omni bench Stage Durations prints those keys (Mean / Median / P99)

## Root cause (code)

`LingBotVideoPipeline` class bases are currently:

```python
class LingBotVideoPipeline(
    nn.Module,
    SupportImageInput,
    ProgressBarMixin,
    SupportsComponentDiscovery,
):
```

There is no `DiffusionPipelineProfilerMixin`, no `setup_diffusion_pipeline_profiler`, and no `stage_durations` on `DiffusionOutput` from this pipeline. `consume_pipeline_stage_durations()` therefore returns `{}` when `enable_diffusion_pipeline_profiler` is unset on the pipeline object, even if the CLI flag was passed into `OmniDiffusionConfig`.

Orchestrator still merges `queue_wait_ms` + `stage_{id}_gen_ms` into `stage_durations`, which is what the bench currently prints.

## Proposed fix (suggestion)

1. Mix in `DiffusionPipelineProfilerMixin` on `LingBotVideoPipeline`.
2. Call `setup_diffusion_pipeline_profiler(...)` from `__init__` gated by `od_config.enable_diffusion_pipeline_profiler`, with targets covering the offline T2V hot path (mirror Wan defaults and/or LingBot-world custom targets as appropriate).
3. Return `stage_durations=self.stage_durations` on `DiffusionOutput`.
4. Add a small unit/e2e check that with the flag enabled, video response `stage_durations` contains at least one pipeline profiler key (not only `queue_wait_ms` / `stage_0_gen_ms`).

## Impact

- Nightly / local DFX LingBot video perf cannot report pipeline stage breakdown.
- Users enabling `--enable-diffusion-pipeline-profiler` for dense LingBot get a silent no-op.

### PR DESCRIPTION
**Pull request by Bezdarnost**
## Purpose

Fixes #7741.

`LingBotVideoPipeline` ignores `--enable-diffusion-pipeline-profiler`, so video responses contain no pipeline-level timings. Wire the pipeline into the existing profiler and include a per-request timing snapshot in `DiffusionOutput`.

The profiler records `text_encoder.forward`, `transformer.forward`, `vae.encode`, and `vae.decode` when those paths execute. Transformer time accumulates across denoising/CFG calls; it is not labelled as the duration of the entire denoising loop. Profiling remains opt-in, and disabled output retains an empty timing dictionary.

## Validation

- LingBot CPU suite: **121 passed**, two GPU tests deselected.
- Runner, CUDA-profiler control, and video-API CPU suites, run separately from the model suite: **169 passed**, eight tests deselected.
- The new enabled-mode regression fails on base `2ab5d170` because pipeline timings are absent; disabled-mode coverage passes on base. The test exercises constructor wiring, repeated transformer calls, request resets, latent-only output, timing snapshots and consumption.
- All applicable pre-commit hooks passed, including Ruff and mypy.
- RTX 5080, dense 1.3B checkpoint: two T2V requests with profiling disabled and two enabled; 320×192, nine frames, two steps, guidance 3, seed 42. All video tensors are finite and bit-identical across all four runs. Enabled runs report encoder/transformer/VAE timings; disabled runs report none.

The combined model+runner test invocation has one pre-existing logging-capture failure (`test_refresh_cache_without_request_or_pipeline_default_warns`), reproduced on the base with the same test ordering. The separated suites above pass. The local environment emits a vLLM/Omni version mismatch warning.

Environment: Python 3.12.11, PyTorch 2.13.0+cu130, CUDA 13.0, vLLM 0.29.0, Transformers 5.14.1, Diffusers 0.40.0; Omni base `2ab5d170`. No MoE, multi-GPU, or NPU validation; no speedup claim.

Live HTTP verification: two `/v1/videos` requests completed with positive encoder, transformer and VAE timings in the final JSON responses. Both downloaded MP4s decode to nine 320×192 frames. Serving used `--enable-cpu-offload` because the fixed 512×512 image-conditioned startup warmup OOMed without it on the 16 GB card; the direct T2V runs above used no offload.

## CI routing

The regression extends the existing `tests/diffusion/models/lingbot_video/test_pipeline_lingbot_video.py` module with `core_model`, `cpu`, and `diffusion` marks. The ready pipeline already collects it via `tests/diffusion -m 'core_model and cpu'`; no Buildkite change is needed.

AI assistance: implementation and validation were prepared with Codex.

### CODE REVIEW

**Comment by Bezdarnost** [comment]
> This PR appears to belong to: **docs/design/module/diffusion/offloader.md**, **docs/design/module/diffusion/diffusion_model_integration.md**, **docs/design/module/diffusion/index.md**, **docs/design/module/profiling.md**.
> 
> Module owners: @wtomin @RuixiangMa @david6666666
> 
> Routing: @wtomin via module of the changed files, module named in the PR description, semantic router, CODEOWNERS; @RuixiangMa via module of the changed files, module named in the PR description, semantic router; @david6666666 via module of the changed files, module named in the PR description
> 
> @Bezdarnost, please review your own changes and leave a short self-review comment describing what you checked. PRs without author self-review may not be assigned a reviewer.
> 
> Please take a look when you have a chance. If you would like an automated review, mention @vllm-omni-review-bot in a comment.

Self-review: checked that profiling is opt-in, records the actual text-encoder, transformer and VAE calls, and resets timings between requests. Verified that latent-only requests do not inherit VAE timings and previous output snapshots remain unchanged.

Validation:
- 121 LingBot CPU tests and 169 runner/API tests passed in separate runs.
- All applicable pre-commit checks passed, including mypy.
- The new enabled-mode regression fails on the unmodified base.
- Four RTX 5080 generations produced bit-identical outputs with profiling enabled and disabled.
- Two live `/v1/videos` requests returned component timings and valid nine-frame MP4s.

The combined-suite logging-capture failure also reproduces on the base branch. Live serving required CPU offload for startup warmup. MoE, multi-GPU and NPU execution were not tested.

**Review by Gaohan123 — APPROVED** [review]
LGTM. Thanks
